File: app-backend/datamodel/src/main/scala/HITLJob.scala
package com.rasterfoundry.datamodel

import io.circe._
import io.circe.generic.JsonCodec
import io.circe.generic.semiauto._

import java.sql.Timestamp
import java.util.UUID

@JsonCodec
final case class HITLJob(
    id: UUID,
    createdAt: Timestamp,
    createdBy: String,
    modifiedAt: Timestamp,
    owner: String,
    campaignId: UUID,
    projectId: UUID,
    status: HITLJobStatus,
    version: Int
)

object HITLJob {

  def tupled = (HITLJob.apply _).tupled

  final case class Create(
      campaignId: UUID,
      projectId: UUID,
      status: HITLJobStatus
  )

  object Create {
    implicit val decCreate: Decoder[Create] = deriveDecoder
  }

}

File: app-hitl/hitl/src/hitl/models/hitl_job.py
from .base import BaseModel


class HITLJob(BaseModel):
    URL_PATH = "/api/hitl-jobs/"

    def __init__(
        self,
        id=None,
        createdAt=None,
        createdBy=None,
        modifiedAt=None,
        owner=None,
        campaignId=None,
        projectId=None,
        status=None,
        version=None
    ):
        self.id = id
        self.createdAt = createdAt
        self.createdBy = createdBy
        self.modifiedAt = modifiedAt
        self.owner = owner
        self.campaignId = campaignId
        self.projectId = projectId
        self.status = status
        self.version = version

    def to_dict(self):
        return dict(
            id=self.id,
            createdAt=self.createdAt,
            createdBy=self.createdBy,
            modifiedAt=self.modifiedAt,
            owner=self.owner,
            campaignId=self.campaignId,
            projectId=self.projectId,
            status=self.status,
            version=self.version
        )

    def update_job_status(self, status):
        self.status = status
        return self.update()

    @classmethod
    def from_dict(cls, d):
        return cls(
            id=d.get("id"),
            createdAt=d.get("createdAt"),
            createdBy=d.get("createdBy"),
            modifiedAt=d.get("modifiedAt"),
            owner=d.get("owner"),
            campaignId=d.get("campaignId"),
            projectId=d.get("projectId"),
            status=d.get("status"),
            version=d.get("version"),
        )

